### DISCUSSION TITLE
Relation Extraction model: How to predict on new data?

### DISCUSSION THREAD

**Discussion by harshpreetskychub** [explosion/spaCy#12724] [Help: Coding & Implementations]
Hi,
I have trained a model using this [video](https://www.youtube.com/watch?v=8HL-Ap5_Axo). I want to perform some inference on new unseen data. 

I have made all the necessary imports

```python
import spacy
import typer
from pathlib import Path
from spacy.tokens import DocBin, Doc

# make the factory work
from rel_pipe import make_relation_extractor, score_relations

# make the config work
from rel_model import create_relation_model, create_classification_layer, create_instances, create_tensors
```

This is the main code

```python
def main(trained_pipeline: Path):
    nlp = spacy.load(trained_pipeline)
  text = "In addition, the expression of latent membrane protein-1 led to activation of NF-kappaB that was capable of binding the IL-1beta promoter."
      doc = nlp(text)
  
      for ent in doc.ents:
          print(ent.text,' :: ', ent.label_)
```

This is the driver
```python
if __name__ == "__main__":
    typer.run(main)
```

I am getting the following as output

```bash
ℹ Could not determine any instances in doc - returning doc as is.
```

How can I create an inference script using which I can predict on custom text?

**Comment by kadarakos** [comment]
Hey harshpreetskychub,

This error message tells you that there were no entities found: https://github.com/explosion/projects/blob/v3/tutorials/rel_component/scripts/rel_pipe.py#LL77C23-L77C85. 

I would suggest as a first step in debugging to first isolate the `ner` component, evaluate it on its own, load it in your script and check the predictions on some test documents.
